src/commonMain/kotlin/au/id/micolous/metrodroid/transit/thess/ThessUltralightTransitData.kt

```diff
@@ -0,0 +1,120 @@
+/*
+ * ThessUltralightTransitFactory.kt
+ *
+ * Copyright 2024 apo-mak
+ *
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ */
+package au.id.micolous.metrodroid.transit.thess
+
+import au.id.micolous.metrodroid.card.CardType
+import au.id.micolous.metrodroid.card.ultralight.UltralightCard
+import au.id.micolous.metrodroid.multi.FormattedString
+import au.id.micolous.metrodroid.multi.Localizer
+import au.id.micolous.metrodroid.multi.Parcelize
+import au.id.micolous.metrodroid.multi.R
+import au.id.micolous.metrodroid.time.TimestampFull
+import au.id.micolous.metrodroid.transit.CardInfo
+import au.id.micolous.metrodroid.transit.TransitBalance
+import au.id.micolous.metrodroid.transit.TransitBalanceStored
+import au.id.micolous.metrodroid.transit.TransitCurrency
+import au.id.micolous.metrodroid.transit.TransitData
+import au.id.micolous.metrodroid.transit.TransitIdentity
+import au.id.micolous.metrodroid.transit.TransitRegion
+import au.id.micolous.metrodroid.ui.ListItem
+import au.id.micolous.metrodroid.ui.ListItemInterface
+
+private const val NAME = "ThessCard"
+
+@Parcelize
+data class ThessUltralightTransitData(
+    private val mSerial: String,
+    private val mProductCode: Int,
+    private val mCounter0: Int,
+    private val mTransaction: ThessUltralightTransaction?,
+    private val mIsSingleUse: Boolean
+) : TransitData() {
+
+    override val serialNumber: String
+        get() = mSerial
+
+    override val cardName: String
+        get() = NAME
+
+    override val balance: TransitBalance?
+        get() {
+            // Trips remaining = 65,535 - Counter0
+            val tripsRemaining = 0xFFFF - mCounter0
+            return if (tripsRemaining >= 0) {
+                TransitBalanceStored(TransitCurrency.EUR(0), tripsRemaining)
+            } else {
+                null
+            }
+        }
+
+    override val info: List<ListItemInterface>?
+        get() = listOfNotNull(
+            ListItem("Ticket type",
+                if (mIsSingleUse) "Single-use" else "Multi-trip"),
+            ListItem("Product code", "0x${mProductCode.toString(16)}")
+        )
+
+    companion object {
+        private fun getSerial(card: UltralightCard): String {
+            // UID is in pages 0-1, plus a BCC byte
+            return card.tagId.toHexString()
+        }
```

Comment:
You can do 
private fun getSerial(card: UltralightCard): String = card.tagId.toHexString()